packages/devextreme/js/__internal/ui/list/m_list.edit.ts

```diff
@@ -317,6 +318,19 @@ class ListEdit extends ListBase {
     this.scrollToItem(this.option('focusedElement'));
   }
 
+  _getFlatIndex(): number | undefined {
+    const { selectedIndex } = this.option();
+
+    if (isNumeric(selectedIndex) || !selectedIndex) {
+      return selectedIndex;
+    }
+
+    const $item = this._editStrategy.getItemElement(selectedIndex);
+    const index = this.getFlatIndexByItemElement($item);
```

Comment:
It looks more logically to make this specific processing under `if` condition (we're trying to change only this, right?)  and return usual selectedIndex in all another scenarios